src/lib.rs

```diff
@@ -709,6 +709,259 @@ impl<'text> BidiInfo<'text> {
     }
 }
 
+/// Bidi information of the text (UTF-16 version).
+///
+/// The `original_classes` and `levels` vectors are indexed by code unit offsets into the text.  If a
+/// character is multiple code units wide, then its class and level will appear multiple times in these
+/// vectors.
+// TODO: Impl `struct StringProperty<T> { values: Vec<T> }` and use instead of Vec<T>
+#[derive(Debug, PartialEq)]
+pub struct BidiInfoU16<'text> {
```

Comment:
thought: from a public API point of view, I wonder if this should all go in a `utf16` module.

From a code organization POV it probably should.